File: midpointlistener1.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int16
from geometry_msgs.msg import Twist
def callback(data):
    rospy.loginfo( "%d",data.data)
    a=data.data
    if a==-2:
        pub = rospy.Publisher('/husky/cmd_vel', Twist)
        twist = Twist()
        rospy.loginfo("Moving the robot left.")
        twist.angular.z =-(a/2.5)  # 
        #for i in range(1,10):
        pub.publish(twist)
            #if a==-1 or 0 or 1:
             #   twist=Twist()
              #  pub.publish(twist)
    elif a==-1:
         rospy.loginfo("Moving the robot left.")
         twist.angular.z =-(a/1.5)  # 
         #for i in range(1,6):
         pub.publish(twist)
            #if a==1 or 0 or 2:
             #   twist=Twist()
              #  pub.publish(twist)
    elif a==1:
        rospy.loginfo("Moving the robot right.")
        twist.angular.z =-(a/1.5)  # 
        #for i in range(1,6):
        pub.publish(twist)
           # if a==-1 or 0 or -2:
            #    twist=Twist()
             #   pub.publish(twist)
    elif a==2:
        rospy.loginfo("Moving the robot right.")
        twist.angular.z =-(a/2.5)  # 
        #for i in range(1,10):
        pub.publish(twist)
           # if a==1 or 0 or -1:
            #    twist=Twist()
             #   pub.publish(twist)
    elif a==0:
        rospy.loginfo("Moving the robot straight.")
        twist.linear.x =.8 #
        #for i in range(1,10): 
        pub.publish(twist)    
        
def listener():
    rospy.init_node('midpointlistener', anonymous=True)
    rospy.Subscriber("midpoint", Int16, callback)
    pub = rospy.Publisher('/husky/cmd_vel', Twist)
    twist = Twist()
    while (ros::ok())
	ospy.spin()
        
if __name__ == '__main__':
    listener()
#### this code is so slow on Husky

File: midpointlistener1.cpp
#include "ros/ros.h"
#include "std_msgs/Int16.h"
#include <iostream>
#include <geometry_msgs/Twist.h>
using namespace std;
std_msgs::Int16 msg;
int a=0;
void chatterCallback(const std_msgs::Int16 msg) //this code i should be using
{
a=msg.data;
}
int main(int argc, char **argv)
{
ros::init(argc, argv, "midpointlistener");
ros::NodeHandle n;
ros::Publisher chatter_pub = n.advertise<std_msgs::Int16>("talker", 1);
ros::Subscriber sub = n.subscribe("midpoint", 1, chatterCallback);
 ros:: Publisher vel_pub = n.advertise<geometry_msgs::Twist>("cmd_vel", 1);
geometry_msgs::Twist base_cmd;

while(ros::ok())
{
ros::spinOnce();
ROS_INFO("I heard: %d", a);
base_cmd.linear.x = a/2;
vel_pub.publish(base_cmd);
}
}

